xbmc/utils/XBMCTinyXML2.cpp

```diff
@@ -57,8 +57,8 @@ bool CXBMCTinyXML2::SaveFile(const std::string& filename) const
   {
     tinyxml2::XMLPrinter printer;
     Accept(&printer);
-    bool suc =
-        file.Write(printer.CStr(), printer.CStrSize()) == static_cast<ssize_t>(printer.CStrSize());
+    const ssize_t sizeToWrite = printer.CStrSize() - 1; // strip trailing '\0'
```

Comment:
I'm not sure where the trailing null is coming from, but this seems really generic and might bite us in the ass if the actual cause of the null is resolved later. Better to check it's actually a null as part the process?

Reply:
The trailing null is in the documentation so I'd say we can safely remove it without further checking:
```cpp
    /**
    	If in print to memory mode, return the size
    	of the XML file in memory. (Note the size returned
    	includes the terminating null.)
    */
    int CStrSize() const {
        return _buffer.Size();
    }
```

Reply:
All good then